scipy/stats/_distn_infrastructure.py

```diff
@@ -2212,21 +2239,36 @@ def fit(self, data, *args, **kwds):
               function to be optimized) and ``disp=0`` to suppress
               output as keyword arguments.
 
+            - method : The method to use. The default is "MLE" (Maximum
+              Likelihood Estimate); "GMM" (Generalized Method of Moments)
+              is also available.
+
+            - order : The number of moments to fit with ``method="GMM"``.
+              This integer must be greater than or equal to the number of
```

Comment:
will `integer` here be positive integers only? Generally nothing precludes negative integers except for pesky division by zero errors. Might be easy to mitigate those issues by using only non-negative.

Reply:
I don't really understand this, either. `order` is the number of moments the function will attempt to fit, and it seems to me that it should be greater than or equal to the number of distribution parameters, so I wrote there "This integer must be greater than or equal to the number of  non-fixed distribution parameters (including location and scale)." This would preclude negative integers.

I did not imagine that users would be able to define _which_ moments to fit. I suppose we could do that.

Reply:
>I did not intend to allow the user to choose which moments to fit. You're right, there is no technical reason why the distribution parameters cannot be optimized to best fit, say, only the 2nd, 3rd, and 5th moments. I suppose it would be easy enough to accept a list of numbers corresponding with the moments to fit, if you think that would be useful. In this case, yes, we would still need to require only positive integers because rv_continuous.moment only allows positive integers.

In this case the `g()` function might also solve the same problem by passing in `callable`s for the different moments, then you wouldn't have the restriction on positive integers. This I'm still a little unclear on how the `g()` function will be defined, are those defined as (population) moments and then set equal to (sample) moments and the root finding equations are solved? If so then being able to set non-consecutive arguments would be a nice option but we could always add that later, right? 

If there are specific examples in existing distributions where the setting would make things easier then it may make sense, but none that I can think of right now.

Reply:
I'm not sure if this makes it easier in any specific cases so consider this a non-blocking question :). 
If possible it would be good to preserve the option of adding in the future, if someone thinks it would be useful for more than 1 distribution. Of course overriding the `.fit` method is always the other option.

Reply:
Originally, I thought the "generalized" part was just allowing for more conditions than parameters to fit (which also calls for an appropriate norm weight matrix), but after you mentioned this I see that it also refers to generalizing the notion of a moment.  Of course, this is also resolved by de-scoping, but I agree that a list of callables will be a good idea when this is generalized.

Reply:
>but after you mentioned this I see that it also refers to generalizing the notion of a moment

yeah. 

> Of course, this is also resolved by de-scoping, but I agree that a list of callables will be a good idea when this is generalized.

I like the ~simple~ elegant approach ;)